src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -157,12 +157,18 @@ def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None
         logger.debug(f'Post remove daemon {self.TYPE}.{daemon.daemon_id}')
 
         if 'dashboard' in self.mgr.get('mgr_map')['modules']:
-            # remove config for dashboard iscsi gateways
+            # check if daemon.hostname in gateway list to prevent possible error
+            # when iscsi service is removed before gateway list is updated
             ret, out, err = self.mgr.check_mon_command({
```

Comment:
What about avoiding calling `check_mon_command` in `post_remove` altogether? We know this can fail for any reason . Can we avoid trying to find all the reasons it can fail one after another?

```suggestion
        ret, out, err = self.mgr.mon_command({
```

Reply:
Oh I didn't know about `mon_command` until now, seems to be a better solution, will change to it